### PR TITLE
Frontend With Framework - fixes issue-#1766

### PR DESCRIPTION
**Pull request by CSAURABH**
<hr>

## Description 📜
fixes issue-#1766 : React Props vs State
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes # <#1766> 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Doc#umentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [x] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] Java#Script
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by CSAURABH** [comment]
@anandhu720 please check my pull request.

**Comment by the-pro** [comment]
@CSAURABH Please improve the md formatting. Also properly indent the code you have written.

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+```javascript
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {title: "Morning"};
+  }
+    changeTitle = () => {
+    this.setState({Title : "Night"});
+
```
Comment (mitraditya) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md:184]: This function doesn't have closing braces.

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+  render() {
+    return (
+      <div>
+        <h1>Good {this.state.title}!</h1>
+      </div>
+      <button type="button" onClick={this.changeTitle}>Change Title</button>
```
Comment (mitraditya) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md:190]: Wrap up the jsx components into a single element.

**Review by mitraditya — COMMENTED** [review]
Please make these changes.

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md (hunk 2 above)
Comment (CSAURABH) [reply Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md:190]: @mitraditya okk sir

**Comment by mitraditya** [comment]
Looks good to me.

**Comment by CSAURABH** [comment]
@mitraditya @the-pro @anandhu720  changes done. please review my pr

**Comment by mitraditya** [comment]
Kindly improve the formatting, as @the-pro suggested.

**Comment by CSAURABH** [comment]
@mitraditya changes done, please review my pr.

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    <App 
+        name = "World!!",
+        title = "Props"    
+     />,
+     document.getElementById('root')
+     );
+
+```
+
+we passes the props as attributes and then access them inside our component using props.title, So props makes components reusable by giving components the ability to receive data from the parent component in the form of props. They are immutable.
```
Comment (mitraditya) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md:100]: *We pass the props as...

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    <App 
+        name = "World!!",
+        title = "Props"    
+     />,
+     document.getElementById('root')
+     );
+
+```
+
+A simple component and then we passes the props as attributes and then access them inside our component using this.props. So props makes components reusable by giving components the ability to receive data from the parent component in the form of props. They are immutable.
```
Comment (mitraditya) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md:53]: *and then we pass the ...

File: Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
+<p align="center"><img src="https://cdn-media-1.freecodecamp.org/images/1*QeEbGcmtoQ3T_zUdpG8TZw.jpeg" width="500px"></p><br>
+
+# React Props vs State
+
+## React Props :-
+
+We know that components in React are used again and again in the UI, but we don’t normally render the same component with the same data. Sometimes we need to change the content inside a component. Props come to play in these cases, as they are passed into the component and the user. Let’s see how they work:
```
Comment (mitraditya) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 2/React_Props_vs_State.md:7]: *passed into the component by the user.

**Review by mitraditya — COMMENTED** [review]
Just these minor edits, everything else looks good to go.

**Comment by CSAURABH** [comment]
@mitraditya Done sir, please check.

**Comment by mitraditya** [comment]
Looks good.

**Pull request merged**